src/rust/ide/src/view/node_editor.rs

```diff
@@ -162,72 +162,77 @@ impl GraphEditorIntegratedWithController {
-        let node_entered = Self::ui_action(&model,
-           GraphEditorIntegratedWithControllerModel::node_entered_in_ui,&invalidate.trigger);
+            GraphEditorIntegratedWithModelModel::node_removed_in_ui,&invalidate.trigger);
+        let node_stepped_into = Self::ui_action(&model,
+            GraphEditorIntegratedWithModelModel::node_stepped_into_in_ui,&invalidate.trigger);
+        let node_stepped_out = Self::ui_action(&model,
+            GraphEditorIntegratedWithModelModel::node_stepped_out_in_ui,&invalidate.trigger);
         let connection_created = Self::ui_action(&model,
-            GraphEditorIntegratedWithControllerModel::connection_created_in_ui,&invalidate.trigger);
+            GraphEditorIntegratedWithModelModel::connection_created_in_ui,&invalidate.trigger);
         let connection_removed = Self::ui_action(&model,
-            GraphEditorIntegratedWithControllerModel::connection_removed_in_ui,&invalidate.trigger);
+            GraphEditorIntegratedWithModelModel::connection_removed_in_ui,&invalidate.trigger);
         let node_moved = Self::ui_action(&model,
-            GraphEditorIntegratedWithControllerModel::node_moved_in_ui,&invalidate.trigger);
+            GraphEditorIntegratedWithModelModel::node_moved_in_ui,&invalidate.trigger);
         let visualization_enabled = Self::ui_action(&model,
-            GraphEditorIntegratedWithControllerModel::visualization_enabled_in_ui,
+            GraphEditorIntegratedWithModelModel::visualization_enabled_in_ui,
             &invalidate.trigger);
         let visualization_disabled = Self::ui_action(&model,
-            GraphEditorIntegratedWithControllerModel::visualization_disabled_in_ui,
+            GraphEditorIntegratedWithModelModel::visualization_disabled_in_ui,
             &invalidate.trigger);
         frp::extend! {network
-            // Notifications from controller
+            // Notifications from model
             let handle_notification = FencedAction::fence(&network,
-                f!((notification:&Option<controller::graph::executed::Notification>)
-                    model.handle_controller_notification(notification);
+                f!((notification:&Option<model::synchronized::graph::executed::Notification>)
+                    model.handle_model_notification(notification);
             ));
 
             // Changes in Graph Editor
             let is_handling_notification = handle_notification.is_running;
             def is_hold = is_handling_notification.all_with(&invalidate.is_running, |l,r| *l || *r);
             def _action = editor_outs.node_removed             .map2(&is_hold,node_removed);
-            def _action = editor_outs.node_entered             .map2(&is_hold,node_entered);
+            def _action = editor_outs.node_stepped_into        .map2(&is_hold,node_stepped_into);
+            def _action = editor_outs.node_stepped_out         .map2(&is_hold,node_stepped_out);
```

Comment:
IF youre touching it, please fix it to use new FRP style - just remove "def" and use `<-` instead of `=`.

Reply:
done